extensions/cli-adapter/deployment/src/main/java/io/quarkiverse/mcp/server/cli/adapter/deployment/McpServerCliAdapterProcessor.java

```diff
@@ -0,0 +1,243 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.AnnotationValue;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.IndexView;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkiverse.mcp.server.Tool;
+import io.quarkiverse.mcp.server.ToolArg;
+import io.quarkiverse.mcp.server.cli.adapter.runtime.McpAdapter;
+import io.quarkus.arc.Unremovable;
+import io.quarkus.arc.deployment.AutoAddScopeBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.arc.deployment.GeneratedBeanGizmoAdaptor;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.gizmo.CatchBlockCreator;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.FieldCreator;
+import io.quarkus.gizmo.FieldDescriptor;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.quarkus.gizmo.TryBlock;
+import picocli.CommandLine.Command;
+
+class McpServerCliAdapterProcessor {
+
+    private static final String FEATURE = "mcp-server-cli-adapter";
+
+    private static final String NON_ALPHANUMERIC = "[^a-zA-Z0-9]";
+    private static final String UNDERSCORE = "_";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void createCommandWrappers(
+            CombinedIndexBuildItem combinedIndex,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+            List<AutoAddScopeBuildItem> autoAddScopeBuildItems) {
+
+        IndexView index = combinedIndex.getIndex();
+        for (DotName c : classesAnnotatedWith(index, Command.class.getName())) {
+            ClassInfo beanClass = index.getClassByName(c);
+
+            if (beanClass.name().toString().startsWith(DotNames.COMMANDLINE.packagePrefix())) {
+                continue;
+            }
+
+            AnnotationInstance commandAnnotation = beanClass.annotation(DotNames.COMMAND);
+            String name = beanClass.simpleName().replaceAll(NON_ALPHANUMERIC, UNDERSCORE).toLowerCase();
+            String description = getCommandDescriptionOrHeader(commandAnnotation);
+
+            String beanClassName = beanClass.name().toString();
+            String generatedWrapperClassName = beanClassName + "McpWrapper";
+
+            // Use Gizmo to generate the new class.
```

Comment:
It might make sense to add some comment about how what the generated wrapper does and how it actually looks like. WDYT?